File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] ans = new int[commands.length];

        for (int i = 0; i < commands.length; i++) {
            int[] tmp = Arrays.copyOfRange(array, commands[i][0] - 1, commands[i][1]);
            Arrays.sort(tmp);
            ans[i] = tmp[commands[i][2] - 1];
        }
        return ans;
    }
}

// n-1, m

File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    let ans = [];

    for (let e of commands) {
        let start = e[0] - 1;
        let end = e[1];
        let index = e[2] - 1;

        let arr = array.slice(start, end).sort((o1, o2) => o1 - o2);
        ans.push(arr[index]);
    }

    return ans;
}
